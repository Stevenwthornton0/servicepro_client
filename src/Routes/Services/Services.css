.serviceContainer1, .serviceContainer2 {
    display: flex;
    flex-direction: row;
    margin-bottom: 25px;
}

.serviceContainer1 > a, .serviceContainer2 > a {
    width: 230px;
    height: 230px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-decoration: none;
    color: black;
    border: 2px solid #bdbdbd;
    border-radius: 4px;
    margin-right: 20px;
}

.serviceContainer1 > a:hover, .serviceContainer2 > a:hover {
    background-color: lightblue;
    border: 1px solid lightblue;
}

.serviceContainer1 > a:last-child, .serviceContainer2 > a:last-child {
    margin-right: 0;
}

.serviceContainer1 > a > p, .serviceContainer2 > a > p {
    margin-top: 20px;
    margin-left: auto;
    margin-right: auto;
    font-size: 18px;
}

.logo {
    margin-left: auto;
    margin-right: auto;
    font-size: 50px;
    color: #424242;
}

.services {
    margin-top: 70px;
    width: fit-content;
    margin-left: auto;
    margin-right: auto;
}

@media screen and (max-width: 770px) {
    .services {
        display: flex;
        flex-direction: row;
        width: fit-content;
        margin-top: 30px;
    }

    .serviceContainer1 {
        display: flex;
        flex-direction: column;
    }

    .serviceContainer2 {
        width: 100%;
        display: flex;
        flex-direction: column;
    }

    .serviceContainer1 > a, .serviceContainer2 > a {
        width: 150px;
        height: 150px;
        margin-right: 0;
    }

    .serviceContainer1 > a > p, .serviceContainer2 > a > p {
        margin-bottom: 0px;
    }
}