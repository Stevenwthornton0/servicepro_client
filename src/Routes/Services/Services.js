import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { Link } from 'react-router-dom';
import ServicesContext from '../../contexts/ServicesContext';
import './Services.css';

class Services extends Component {

    static contextType = ServicesContext;

    componentDidMount() {
        const { open } = this.context;
        if (open === true) {
            this.context.setOpen()
        }
    }

    render() {
        return (
            <div className='services'>
                <div className='serviceContainer1'>
                    <Link to={'/services/mechanic'}>
                        <FontAwesomeIcon className='logo car' icon='car' />
                        <p>Mechanic</p>
                    </Link>

                    <Link to={'/services/plumbing'}>
                        <FontAwesomeIcon className='logo toilet' icon='toilet' />
                        <p>Plumbing</p>
                    </Link>

                    <Link to={'/services/remodling'}>
                        <FontAwesomeIcon className='logo home' icon='home' />
                        <p>Home Remodel</p>
                    </Link>

                    <Link to={'/services/lawnscaping'}>
                        <FontAwesomeIcon className='logo mountain' icon='mountain' />
                        <p>Lawnscaping</p>
                    </Link>
                </div>

                <div className='serviceContainer2'>
                    <Link to={'/services/travel'}>
                        <FontAwesomeIcon className='logo plane' icon='plane' />
                        <p>Travel</p>
                    </Link>

                    <Link to={'/services/hair'}>
                        <FontAwesomeIcon className='logo cut' icon='cut' />
                        <p>Hair Stylist</p>
                    </Link>

                    <Link to={'/services/jobs'}>
                        <FontAwesomeIcon className='logo coins' icon='coins' />
                        <p>Employment</p>
                    </Link>

                    <Link to={'/services/phone'}>
                        <FontAwesomeIcon className='logo mobile' icon='mobile' />
                        <p>Phone Repair</p>
                    </Link>
                </div>

            </div>
        )
    }
}

export default Services;