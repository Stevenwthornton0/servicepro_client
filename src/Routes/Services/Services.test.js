import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import Services from './Services';

Enzyme.configure({ adapter: new Adapter() })

describe('Services component', () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<Services />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})