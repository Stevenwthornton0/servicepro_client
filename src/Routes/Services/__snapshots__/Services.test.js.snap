// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services component renders without crashing 1`] = `
<div
  className="services"
>
  <div
    className="serviceContainer1"
  >
    <Link
      to="/services/mechanic"
    >
      <FontAwesomeIcon
        border={false}
        className="logo car"
        fixedWidth={false}
        flip={null}
        icon="car"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <p>
        Mechanic
      </p>
    </Link>
    <Link
      to="/services/plumbing"
    >
      <FontAwesomeIcon
        border={false}
        className="logo toilet"
        fixedWidth={false}
        flip={null}
        icon="toilet"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <p>
        Plumbing
      </p>
    </Link>
    <Link
      to="/services/remodling"
    >
      <FontAwesomeIcon
        border={false}
        className="logo home"
        fixedWidth={false}
        flip={null}
        icon="home"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <p>
        Home Remodel
      </p>
    </Link>
    <Link
      to="/services/lawnscaping"
    >
      <FontAwesomeIcon
        border={false}
        className="logo mountain"
        fixedWidth={false}
        flip={null}
        icon="mountain"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <p>
        Lawnscaping
      </p>
    </Link>
  </div>
  <div
    className="serviceContainer2"
  >
    <Link
      to="/services/travel"
    >
      <FontAwesomeIcon
        border={false}
        className="logo plane"
        fixedWidth={false}
        flip={null}
        icon="plane"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <p>
        Travel
      </p>
    </Link>
    <Link
      to="/services/hair"
    >
      <FontAwesomeIcon
        border={false}
        className="logo cut"
        fixedWidth={false}
        flip={null}
        icon="cut"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <p>
        Hair Stylist
      </p>
    </Link>
    <Link
      to="/services/jobs"
    >
      <FontAwesomeIcon
        border={false}
        className="logo coins"
        fixedWidth={false}
        flip={null}
        icon="coins"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <p>
        Employment
      </p>
    </Link>
    <Link
      to="/services/phone"
    >
      <FontAwesomeIcon
        border={false}
        className="logo"
        fixedWidth={false}
        flip={null}
        icon="mobile"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <p>
        Phone Repair
      </p>
    </Link>
  </div>
</div>
`;
